### DISCUSSION TITLE
Add a Versioning System To YimMenu

### DISCUSSION THREAD

**Discussion by xesdoog** [YimMenu/YimMenu#3218] [Ideas]
Hello beautiful yimmies,

Some devs tried/are trying to make a launchpad for YimMenu: 
- USBMenu's [YimUpdater](https://github.com/USBMenus/Yim-Updater).
- NiiV3AU's [YMU](https://github.com/NiiV3AU/YMU).

Both of these work and both are nice but the problem is they have no way to determine if the user has the latest YimMenu dll or not other than having to download the dll each time, maybe store it in a temporary location, calculate its hash, calculate the hash of the already existing dll and compare the two. This whole tedious process can be completely eliminated by just adding a versioning system either to the dll (I know this will require modifying YimMenu's workflow) or just simply adding a new link to the website similar to the [changelog link](https://yim.gta.menu/changelog.html) that just displays a version number.
On program init, we start by reading that number and check if the user already has a dll downloaded to a path controller by the program. If not, download YimMenu and assign that number as the version the user has (store it in a text file or rename the dll)... You see where I'm going with this?

Thank you for coming to my TEDTalk

**Comment by Yimura** [comment]
Those launchers just need to maintain an internal value to keep track of the latest hash and compare it against the remote hash. If it’s different download a new dll.

**Reply by xesdoog** [reply]
Sorry for asking a noob question but the remote hash you mentioned is the one displayed in the Github release tab?

**Reply by Rxann** [reply]
You could use something like shown here: https://stackoverflow.com/questions/1220046/md5-hash-from-file-in-c

**Reply by xiaoxiao921** [reply]
> You could use something like shown here: https://stackoverflow.com/questions/1220046/md5-hash-from-file-in-c

no because that imply downloading the dll everytime and making a hash out of it, what you need is just storing the latest commit hash linked to the nightly release

**Reply by Rxann** [reply]
> > You could use something like shown here: https://stackoverflow.com/questions/1220046/md5-hash-from-file-in-c
> 
> 
> 
> no because that imply downloading the dll everytime and making a hash out of it, what you need is just storing the latest commit hash linked to the nightly release

Ahh, I see. That makes more sense 🤣. Thanks

**Reply by xesdoog** [reply]
> what you need is just storing the latest commit hash linked to the nightly release

Guys, I keep opening discussions to things that already exist 😭 This is embarrassing lol
